demo/angular/gridstack.component.ts

```diff
@@ -0,0 +1,280 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChildren,
+  ElementRef,
+  EventEmitter,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+  QueryList,
+  Renderer2,
+} from '@angular/core';
+import {GridstackItemComponent} from './gridstack-item.component';
+import {merge, Subject} from 'rxjs';
+import {GridStack, GridStackEvent, GridStackOptions} from 'gridstack';
+import 'gridstack/dist/h5/gridstack-dd-native';
+import {map, takeUntil} from 'rxjs/operators';
+
+@Component({
+  selector: 'ef-gridstack',
+  templateUrl: './gridstack.component.html',
+  styleUrls: ['./gridstack.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GridstackComponent implements OnInit, OnDestroy {
+
+  @ContentChildren(GridstackItemComponent) public gridstackItems: QueryList<GridstackItemComponent>;
+
+  @Input() public options: GridStackOptions;
+
+  @Output() public gridstackAdded = new EventEmitter<{
```

Comment:
those should all be typed since they are the same - not clear why you need to pass Gridstack again (caller would what that).

Reply:
The caller might need the grid reference to get all the current elements and their position in it to save the user's customization. That's why I'm doing in my app.